packages/core/notarization/manager.go

```diff
@@ -182,27 +183,20 @@ func (m *Manager) LoadECandEIs(header *ledger.SnapshotHeader) {
 	m.epochCommitmentFactory.storage.ecRecordStorage.Store(header.LatestECRecord).Release()
 }
 
-// LoadSnapshot initiates the state and mana trees from a given snapshot.
-func (m *Manager) LoadSnapshot(s *ledger.Snapshot) {
-	m.LoadOutputWithMetadatas(s.OutputsWithMetadata)
-	m.LoadEpochDiffs(s.Header, s.EpochDiffs)
-	m.LoadECandEIs(s.Header)
-}
-
+// SnapshotEpochDiffs returns the EpochDiffs when a snapshot is created.
 func (m *Manager) SnapshotEpochDiffs() (map[epoch.Index]*ledger.EpochDiff, error) {
-	m.epochCommitmentFactoryMutex.Lock()
-	defer m.epochCommitmentFactoryMutex.Unlock()
-
 	start, err := m.LatestConfirmedEpochIndex()
 	if err != nil {
 		return nil, err
 	}
 
-	ec, err := m.GetLatestEC()
+	m.epochCommitmentFactoryMutex.Lock()
```

Comment:
This should be an `RLock`. You can also place it at the beginning of the method, as you are also reading the LatestConfirmedIndex